pkg/logging/logging.go

```diff
@@ -81,16 +86,60 @@ func InitializeDefaultLogger() *logrus.Logger {
 	return logger
 }
 
+// GetLogLevelFromConfig returns the log level provided via global
+// configuration. If the logging level is invalid, ok will be false.
+func GetLogLevelFromConfig() (logrus.Level, bool) {
+	return options.GetLogLevel()
+}
+
+// GetLogLevel returns the log level specified in the provided LogOptions. If
+// it is not set in the options, ok will be false.
+func (o LogOptions) GetLogLevel() (logrus.Level, bool) {
+	lvl, ok := LevelStringToLogrusLevel[strings.ToLower(o[LevelOpt])]
+	return lvl, ok
+}
+
+// configureLogLevelFromOptions returns the log level based off of the value of
+// LevelOpt in o, or the default log level if the value in the map is invalid or
+// not set.
+func (o LogOptions) configureLogLevelFromOptions() logrus.Level {
+	var level logrus.Level
+	if levelOpt, ok := o[LevelOpt]; ok {
+		if convertedLevel, ok := o.GetLogLevel(); ok {
+			level = convertedLevel
+		} else {
+			// Invalid configuration provided, go with default.
+			DefaultLogger.WithField(logfields.LogSubsys, "logging").Warningf("invalid logging level provided: %s; setting to %s", levelOpt, DefaultLogLevelStr)
+			o[LevelOpt] = DefaultLogLevelStr
+			level = LevelStringToLogrusLevel[DefaultLogLevelStr]
+		}
+	} else {
+		// No logging option provided, default to DefaultLogLevelStr.
+		o[LevelOpt] = DefaultLogLevelStr
+		level = LevelStringToLogrusLevel[DefaultLogLevelStr]
+	}
+	return level
+}
+
+// configureLogLevelFromOptions sets the log level of the DefaultLogger based
+// off of the value of LevelOpt in logOpts. If LevelOpt is not set in logOpts,
+// it defaults to DefaultLogLevelStr.
+func setLogLevelFromOptions(logOpts LogOptions) {
```

Comment:
It feels a little odd to have this private function but a global GetLogLevel that is more generic. Creating a `SetLogLevel(logrus.Level)` feels more re-usable. It is just shuffling things around, though, since we only use this in one callpath.